src/clientrpc.cpp

```diff
@@ -0,0 +1,201 @@
+            };
+            arrChanInfo.append ( objChanInfo );
+        }
+        pRpcServer->BroadcastNotification ( "jamulusclient/clientListReceived",
+                                            QJsonObject{
+                                                { "clients", arrChanInfo },
+                                            } );
+        arrStoredChanInfo = arrChanInfo;
+    } );
+
+    connect ( pClient, &CClient::CLChannelLevelListReceived, [=] ( CHostAddress /* unused */, CVector<uint16_t> vecLevelList ) {
+        QJsonArray arrLevelList;
+        for ( const auto& level : vecLevelList )
+        {
+            arrLevelList.append ( level );
+        }
+        pRpcServer->BroadcastNotification ( "jamulusclient/channelLevelListReceived",
+                                            QJsonObject{
+                                                { "channelLevelList", arrLevelList },
+                                            } );
+    } );
+
+    connect ( pClient, &CClient::Disconnected, [=]() { pRpcServer->BroadcastNotification ( "jamulusclient/disconnected", QJsonObject{} ); } );
+
+    pRpcServer->HandleMethod ( "jamulus/getMode", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        QJsonObject result{ { "mode", "client" } };
+        response["result"] = result;
+        Q_UNUSED ( params );
+    } );
+
+    pRpcServer->HandleMethod ( "jamulusclient/getClientInfo", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        QJsonObject result{ { "connected", pClient->IsConnected() } };
+        response["result"] = result;
+        Q_UNUSED ( params );
+    } );
+
+    pRpcServer->HandleMethod ( "jamulusclient/getChannelInfo", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        QJsonObject result{
+            { "name", pClient->ChannelInfo.strName },
+            { "skillLevel", SerializeSkillLevel ( pClient->ChannelInfo.eSkillLevel ) },
+        };
+        response["result"] = result;
+        Q_UNUSED ( params );
+    } );
+
+    pRpcServer->HandleMethod ( "jamulusclient/getClientList", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        if ( !pClient->IsConnected() )
+        {
+            response["error"] = CRpcServer::CreateJsonRpcError ( 1, "Client is not connected" );
+            return;
+        }
+
+        QJsonObject result{
+            { "clients", arrStoredChanInfo },
+        };
+        response["result"] = result;
+        Q_UNUSED ( params );
+    } );
+
+    pRpcServer->HandleMethod ( "jamulusclient/setName", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        auto jsonName = params["name"];
+        if ( !jsonName.isString() )
+        {
+            response["error"] = CRpcServer::CreateJsonRpcError ( CRpcServer::iErrInvalidParams, "Invalid params: name is not a string" );
+            return;
+        }
+
+        pClient->ChannelInfo.strName = jsonName.toString().left ( MAX_LEN_FADER_TAG );
```

Comment:
There should be a `pClient->setChannelName` or similar that does the truncation, used by the GUI.  The same method should be called here.  If it doesn't exist, it should be created and the GUI amended to use it.

Reply:
Probably QString.left() is concatenating the length. I still don't fully understand why the name didn't change for me.

Reply:
@pljones Can you help perform that refactoring in a separate PR? I am avoiding making changes to existing files in this PR to reduce the chance of conflict.

Otherwise maybe this can be fixed after merging (since this is still “experimental”, imo we only need to cover the happy case).

Reply:
An issue can be opened and tagged to the next release

Reply:
The relevant code from the GUI is here:
https://github.com/jamulussoftware/jamulus/blob/6815cadb1e91fc3d3a217b69664f6b25219c2e7d/src/clientsettingsdlg.cpp#L1082-L1098

I've updated this PR to use `TruncateString()` instead of `left()`in 5d354c87 in order to avoid re-introducing #1994 et al.
I've opened #2438 to track the planned refactoring (`setChannelName`).

> If I set a wrong size, it still returns ok and doesn't seem to change anything

Returning OK is expected based on the code. As it resembles what the UI does, I think it's ok.
Setting a overlong name works for me, it's properly truncated.
@ann0see Where did you see that issue? GUI/Non-GUI client? Is it still reproducible?

Reply:
@hoffie confirmed that the concat now works on the headless client also.

Reply:
@ann0see Thanks
@pljones As the actual truncation should be correct now and your refactoring suggestion is tracked in #2438, I'm marking this thread as resolved.